.github/workflows/main.yml

```diff
@@ -109,7 +107,7 @@ jobs:
             sudo apt-get install valgrind
             valgrind --suppressions=.valgrind-suppress.supp --leak-check=full --gen-suppressions=all \
               --track-origins=yes --error-exitcode=1 build/out/osqp_tester
-          if: ${{ runner.os == 'Linux' && matrix.cmake_flags == '-DUNITTESTS=ON -DCOVERAGE=ON' && matrix.cmake_flags_extra == '-DENABLE_MKL_PARDISO=OFF' }}
+          if: ${{ runner.os == 'Linux' && matrix.cmake_flags == '-DUNITTESTS=ON -DCOVERAGE=ON' && matrix.cmake_flags_extra == '-DALGEBRA_MKL=OFF' }}
```

Comment:
This is likely incorrect now, since CI behavior should be driven off the string `ALGEBRA` constant and not the boolean `ALGEBRA_*` constants. However, I'm not sure if we should run `valgrind` just for the default case or not..

Reply:
Maybe we can even remove the string `ALGEBRA` option; see the discussion below.

Reply:
Also, we probably want to replace `-DALGEBRA_MKL=OFF` by `-DALGEBRA_DEFAULT=ON` in the code snippet above.

Reply:
I think going with the string option `ALGEBRA` will keep the code more future-proof in case we add any more options? The booleans can be derived inside `CMakeLists`, and this will also avoid the problem of trying to detect conflicting flags (see my comment below). So perhaps we can change this line to `-DALGEBRA=default` instead?

Reply:
Sounds good.

Reply:
Done